intranet/static/js/schedule.js

```diff
@@ -100,7 +106,19 @@ $(function() {
     }
 
     getPeriodElem = function(period) {
-        return $(".schedule-block[data-block-order='" + period.order + "']");
+        var elems = $(".schedule-block[data-block-order='" + period.order + "']");
+        if(elems.length > 1) {
+            //We're probably on the calendar view; try and find today's block specifically
+            var dateString = period.end.date.dateFormat("Y-m-d");
+            for(var i = 0; i < elems.length; i++) {
+                var elem = $(elems.get(i));
```

Comment:
Instead of doing these string comparisons, you could just check if the element's grandparent has the today class. https://github.com/tjcsl/ion/blob/1868ef008729dc87fe2e88f9e83ab833b3106331/intranet/templates/schedule/month.html#L11

Reply:
@theo-o That would make . I realize not many people leave this page open overnight and expect it to be accurate the next morning, but this method handles that case and it isn't _that_ bad.

Reply:
@theo-o That would make the page inaccurate when left open past midnight. I realize not many people leave this page open overnight and expect it to be accurate the next morning (and it isn't, since styles are applied to mark the current day based on the `today` class), but this method handles that case and it isn't _that_ bad.

Reply:
Are we expected to guarantee a user (who leaves the page open overnight) that the schedule will even be correct one for the next day?

Reply:
No, but I'm erring on the side of making that the case for the parts I'm changing. The schedule won't be accurate, but it will be partially accurate.